### PR TITLE
Fix #2804: Text Platform roundtrip (with tests)

### LINKED ISSUE #2804: Text instance for Text.System.Platform can't do a round trip
Is that intentional?

~~~~ haskell
λ> simpleParse (display (Platform X86_64 Linux)) :: Maybe Platform
Nothing
~~~~

### PR DESCRIPTION
**Pull request by phadej**

### CODE REVIEW

**Comment by 23Skidoo** [comment]
LGTM.

**Comment by phadej** [comment]
Have to fix the ambiguous parse though.
- Most obvious change would be to not support `-` dash in ident.

otherwise:
- `foo-bar-quux` is indeed ambiguous `Platform`

**Comment by phadej** [comment]
ping @23Skidoo, now this is green. AFAICS parse part of `Text Platform` isn't used in `Cabal` itself, so there shouldn't be any regressions.

File: Cabal/Distribution/System.hs [hunk 1]
```diff
@@ -193,8 +209,14 @@ buildPlatform = Platform buildArch buildOS
 -- Utils:
 
 ident :: Parse.ReadP r String
-ident = Parse.munch1 (\c -> Char.isAlphaNum c || c == '_' || c == '-')
-  --TODO: probably should disallow starting with a number
+ident = liftM2 (:) first rest
```
Comment (23Skidoo) [inline Cabal/Distribution/System.hs]: Looks like `ident` is now identical to `dashlessIdent`. Is this intentional?

Comment (phadej) [inline Cabal/Distribution/System.hs]: My bad, `ident` should have dash. Will fix shortly.

**Comment by phadej** [comment]
@23Skidoo does it look ok now?

**Comment by 23Skidoo** [comment]
It does! Merging, thanks!

**Pull request merged**